### DISCUSSION TITLE
error. No space left on device.

### DISCUSSION THREAD

**Discussion by klnn** [Homebrew/discussions#2667] [Casks] [answered]
New to homebrew. I was trying to 'kicad'. The package was downloaded successfully. Then it immediately shows a list of 'No space left on device' in white, followed by

==> Purging files for version 5.1.12-1 of Cask kicad
Error: Failure while executing; `ditto --bom /private/tmp/20211221-92600-1117dim.bom -- /private/tmp/d20211221-92600-1ss6ghp/dmg.u1vt3G /private/tmp/d20211221-92600-1cxiqz9` exited with 1. Here's the output:

Then it shows what appears to be the same list of files but in red. The package isn't installed. It's not shown in the installed formulae or casks. I tried brew clean up. It doesn't free up any space.

The downloaded files in /library/Caches/Homebrew are rougly 10g, but I think I had more free space than that. I am not sure what to do now. What am I missing? Many thanks.

**Comment by SMillerDev** [comment]
How much free space does macOS say you have?

**Reply by klnn** [reply]
It says 5.77g available now. I usually have 30-40g space.

**Reply by neutric** [reply]
The .dmg is 1.3 GB according to the download page. The app bundle inside is likely going to up at least as much space, which would leave you with roughly 4.4 GB. With this little space left I wouldn't bank on all information displayed by Finder being up to date. If your machine has only 8 GB RAM (guessing by the limited amount of disk space you seem to have) and you are doing more serious work, then it most likely is swapping RAM in and out constantly and performance is degraded.

**Reply by klnn** [reply]
I see. Thanks for the help. I wonder if the package was partially installed somewhere that is not recognised by homebrew. What does the error mean? Why is there a repeat of 'no space' fails in red?

https://ss64.com/osx/ditto.html
I don't know if I understand correctly. Some error occured, presumably due to lack of space, while copying directories, then mission aborted. Does 'existed with 1 ' mean no file has been transferred?

**Reply by synkathairo** [reply]
try running `brew cleanup` to remove cached packages

**Comment by neutric** [comment, marked as answer]
> What does the error mean?

According to `man ditto`, ditto returns 0 if everything is copied, otherwise non-zero. You got 1, so not everything was copied.

> Some error occured, presumably due to lack of space, while copying directories, then mission aborted.

Yes.

>  I wonder if the package was partially installed somewhere that is not recognised by homebrew.

I would assume that the step `==> Purging files for version 5.1.12-1 of Cask kicad` deletes the remnants of non-successful installs. And if it didn't, what good would it do you? `/private/tmp/d20211221-92600-1cxiqz9`, where that ditto run was copying to, is not a permanent folder. It gets cleaned out by the system according to some algorithm, maybe "wasn't accessed in so many days". Also, software with missing parts isn't going to deliver a satisfactory user experience, if you can get it to start up at all.

> Why is there a repeat of 'no space' fails in red?

Your initial post is lacking the promised output, so I can't comment on that.

> It says 5.77g available now. I usually have 30-40g space.

I would focus on that. Find out what occupies the normally available space and free it up so that you can successfully install kicad. There are GUI programs such as `Disk Inventory X`(install with `brew install disk-inventory-x`) or, if you fancy the command line, programs such as `dust`(install with `brew install dust`).

Feel free to ask further questions. If satisfied, please mark this as answered.

**Reply by klnn** [reply]
Thanks a lot my friend :)
